test/check-application

```diff
@@ -15,15 +15,65 @@ import testlib
 # Nondestructive tests all run in the same running VM. This allows them to run in Packit, Fedora, and RHEL dist-git gating
 # They must not permanently change any file or configuration on the system in a way that influences other tests.
 @testlib.nondestructive
-class TestApplication(testlib.MachineCase):
+class TestNavigator(testlib.MachineCase):
     def testBasic(self):
         b = self.browser
         m = self.machine
 
+        self.restore_dir("/home/admin")
+
         self.login_and_go("/navigator")
-        # verify expected heading
+        # expected heading
         b.wait_text(".pf-c-card__title", "Directories & files")
 
+        # new files are auto-detected
+        m.execute("touch /home/admin/newfile")
+        b.wait_visible("[data-item='newfile']")
+
+        # new directories are auto-detected
+        m.execute("mkdir /home/admin/newdir")
+        b.wait_visible("[data-item='newdir']")
+
+        # hidden files are not displayed
+        m.execute("touch /home/admin/.hiddenfile")
+        b.wait_not_present("[data-item='.hiddenfile']")
+
+        # filtering works
+        self.browser.wait_js_cond("ph_count('#folder-view > div') > 1")
+        b.set_input_text("input[placeholder='Filter directory']", "newfile")
+        self.browser.wait_js_cond("ph_count('#folder-view > div') == 1")
+
+        # filtering persists when changing view
+        b.click("button[aria-label='Display as a list']")
+        self.browser.wait_js_cond("ph_count('#folder-view tbody tr') == 1")
+        b.set_input_text("input[placeholder='Filter directory']", "")
+        self.browser.wait_js_cond("ph_count('#folder-view tbody tr') > 1")
+
+        # deleted files and directories are auto-detected
+        m.execute("rmdir /home/admin/newdir && rm /home/admin/newfile")
```

Comment:
Please don't use `&&`. m.execute() is already running under `set -e`, and the semantics of `&&` are really weird.